src/GafferCycles/IECoreCyclesPreview/Renderer.cpp

```diff
@@ -1102,6 +1104,18 @@ class CyclesAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 				ShaderAssignPair pair = ShaderAssignPair( light, ccl::array<ccl::Node*>() );
 				pair.second.push_back_slow( m_lightShader->shader() );
 				m_shaderCache->addShaderAssignment( pair );
+
+				bool isMuted = m_muteLight.value_or( false );
+				if( light->get_is_enabled() != !isMuted )
+				{
+					light->set_is_enabled( !isMuted );
+					if( isMuted )
+					{
+						// Alas, `ccl::LightManager::test_enabled_lights()` will
+						// re-enable the light unless we also set its strength to zero.
+						light->set_strength( ccl::zero_float3() );
+					}
```

Comment:
Rather than repeat this workaround, can we do it once at the bottom of this function, with the test being `if( !light->get_is_enabled()`, to cover both cases?

Reply:
Fixed in 2a515fc072bed38621490420662667d98e32d6ff